o2a/mappers/action_mapper.py

```diff
@@ -24,22 +25,45 @@
 
 # pylint: disable=abstract-method
 # noinspection PyAbstractClass
-class ActionMapper(BaseMapper):
+from o2a.o2a_libs.property_utils import PropertySet
+
+
+class ActionMapper(BaseMapper, ABC):
     """Base class for all action mappers"""
 
-    def __init__(self, oozie_node: Element, name: str, trigger_rule: str = TriggerRule.ALL_SUCCESS, **kwargs):
-        super().__init__(oozie_node, name, **kwargs)
-        self.properties: Dict[str, str] = {}
-        self.trigger_rule: str = trigger_rule
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        job_properties: Dict[str, str],
+        configuration_properties: Dict[str, str],
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        **kwargs: Any,
+    ):
+        super().__init__(
+            oozie_node, name, dag_name, job_properties, configuration_properties, trigger_rule, **kwargs
+        )
+        self.action_node_properties: Dict[str, str] = {}
+        self._parse_config()
 
     def _parse_config(self):
         config = self.oozie_node.find("configuration")
         if config:
+            property_set = self.property_set
             property_nodes = xml_utils.find_nodes_by_tag(config, "property")
```

Comment:
This logic is repeated in several places.I think we should create an helper method - extract_properties_from_configuration. We can do it in seperate PR.

Reply:
No need for that. In fact it was exactly repeated in one place only - in subworkflow_mapper which was completely not needed as it already derives from action mapper.

I removed it in 554a0d9bf56df6123891998b81df35b7684b106f